eth/tracers/logger/logger.go

```diff
@@ -347,3 +370,64 @@ func (t *mdLogger) CaptureEnter(typ vm.OpCode, from common.Address, to common.Ad
 }
 
 func (t *mdLogger) CaptureExit(output []byte, gasUsed uint64, err error) {}
+
+// ExecutionResult groups all structured logs emitted by the EVM
+// while replaying a transaction in debug mode as well as transaction
+// execution status, the amount of gas used and the return value
+type ExecutionResult struct {
+	Gas         uint64         `json:"gas"`
+	Failed      bool           `json:"failed"`
+	ReturnValue string         `json:"returnValue"`
+	StructLogs  []StructLogRes `json:"structLogs"`
+}
```

Comment:
So... we have `logger.ExecutionResult` and `core.ExecutionResult`...And also an `ethapi.ExecutionResult` ?
I don't understand why getresult was added to loggers, actually?

Reply:
No `ethapi.ExecutionResult` anymore. I moved that to `logger.go` because that's the only place it was used. We could further make it private except some tests in `eth/tracers/api_test.go` use it.

My goal was to make the `StructLogger` compliant with the `Tracer` interface. Then the `traceTx` logic becomes even simpler.